File: java/armstrong-numbers/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

	boolean isArmstrongNumber(int numberToCheck) {

		String digits = String.valueOf(numberToCheck);
		int sum = 0;
		int num_digits = digits.length();
		for (int i = 0; i < num_digits; i++)
			sum += Math.pow(Character.getNumericValue(digits.charAt(i)), num_digits);

		return sum == numberToCheck;

	}

}

File: csharp/armstrong-numbers/ArmstrongNumbers.cs
using System;

public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number)
    {
        int numberOfDigits = number > 0 ? (int) Math.Log10(number) + 1 : 1;
        int sum = 0;
        for (int i = number; i > 0; i /= 10)
            sum += (int) (Math.Pow(i % 10, numberOfDigits) + 0.5);

        return sum == number;
    }
}